module/metrics.go

```diff
@@ -424,3 +424,29 @@ type PingMetrics interface {
 	// NodeInfo tracks the software version, sealed height and hotstuff view of a node
 	NodeInfo(node *flow.Identity, nodeInfo string, version string, sealedHeight uint64, hotstuffCurView uint64)
 }
+
+type HeroCacheMetrics interface {
+	// BucketAvailableSlots keeps track of number of available slots in buckets of cache.
+	BucketAvailableSlots(uint64, uint64)
+
+	// OnSuccessfulWrite is called whenever a new entity is successfully added to the cache.
+	OnSuccessfulWrite()
+
+	// OnEntityEjectedAtFullCapacity is called whenever adding a new entity to the cache results in ejection of another entity.
+	// This normally happens when the cache is full.
+	OnEntityEjectedAtFullCapacity()
+
+	// OnEmergencyKeyEjection is called whenever a bucket is found full and all of its keys are valid.
+	// Hence, adding a new entity to that bucket will replace the oldest valid key inside that bucket.
```

Comment:
I've noticed a couple of comments where the first reference to the `key` is called `entity` and the second reference to the `key` is called `key`. I imagine this makes things more confusing for some readers? What about consistently referring to keys as entities or vice-versa? Or whenever talking about a key as an entity then always refer to it as e.g. `entity (i.e. key in heroCache terminology)`?

So this sentence might be the following:

`Hence, adding a new entity [1] to that bucket will replace the oldest valid entity [1] inside that bucket.`
`[1] note: entity is a key in heroCache terminology`